website/docs/d/datasource_compute_instance.html.markdown

```diff
@@ -0,0 +1,187 @@
+---
+layout: "google"
+page_title: "Google: google_compute_instance"
+sidebar_current: "docs-google-datasource-compute-instance-x"
+description: |-
+  Get a VM instance within GCE.
+---
+
+# google\_compute\_instance
+
+Get a VM instance resource within GCE. For more information see
+[the official documentation](https://cloud.google.com/compute/docs/instances)
+and
+[API](https://cloud.google.com/compute/docs/reference/latest/instances).
+
+
+## Example Usage
+
+```hcl
+data "google_compute_instance" "appserver" {
+	name = "primary-application-server"
+	zone = "us-central1-a"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the instance.
```

Comment:
Could we allow retrieval by self_link, as well? I think that'll be a highly requested use case.

Reply:
Sure!

To do this I took the logic from google_compute_region_instance_group's data source and parameterised it for consistency between the two. The test for that data source will fail but it was failing before and fails in the same way. Sorry for the scope of changes as a result of touching both resources here in this already large PR!